File: python/src/cassandragargoyle/api/telemetry/exporter_type.py
# This file is part of CassandraGargoyle Community Project
# Licensed under the MIT License - see LICENSE file for details

from enum import Enum


class ExporterType(Enum):
    """Supported OpenTelemetry exporter types."""

    NONE = "none"
    """No-op exporter, zero overhead"""

    CONSOLE = "console"
    """Console/logging exporter for development"""

    OTLP = "otlp"
    """OTLP exporter for production use"""

File: java/src/main/java/org/cassandragargoyle/api/telemetry/ExporterType.java
/**
 * This file is part of CassandraGargoyle Community Project
 * Licensed under the MIT License - see LICENSE file for details
 */
package org.cassandragargoyle.api.telemetry;

/**
 * Supported OpenTelemetry exporter types
 *
 * @author Zdenek
 * @since 2026-03-18
 */
public enum ExporterType
{
	/** No-op exporter, zero overhead */
	NONE,

	/** Console/logging exporter for development */
	CONSOLE,

	/** OTLP exporter for production use */
	OTLP
}
